src/main/java/io/mathdojo/QuestionHandler.java

```diff
@@ -0,0 +1,71 @@
+package io.mathdojo;
+
+import java.util.Optional;
+
+import org.springframework.cloud.function.adapter.azure.AzureSpringBootRequestHandler;
+
+import com.microsoft.azure.functions.ExecutionContext;
+import com.microsoft.azure.functions.HttpMethod;
+import com.microsoft.azure.functions.HttpRequestMessage;
+import com.microsoft.azure.functions.annotation.AuthorizationLevel;
+import com.microsoft.azure.functions.annotation.BindingName;
+import com.microsoft.azure.functions.annotation.FunctionName;
+import com.microsoft.azure.functions.annotation.HttpTrigger;
+
+/**
+ * 
+ * This class handles http requests specific to questions
+ *
+ */
+
+public class QuestionHandler extends AzureSpringBootRequestHandler<Question, Question> {
+
+	@FunctionName("getQuestion")
+	public Question executeGet(@HttpTrigger(name = "request", methods = {
+			HttpMethod.GET }, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<Question>> request,
+			ExecutionContext context) {
+		context.getLogger().info("Retrieving question");
+
+		return handleRequest(new Question(request.getQueryParameters()), context);
+	}
+
+	@FunctionName("createQuestion")
+	public Question execute(@HttpTrigger(name = "request", methods = {
+			HttpMethod.POST }, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<Question>> request,
+			ExecutionContext context) {
+		context.getLogger().info("Posting question with title: " + request.getBody().get().getQuestionTitle());
+		return handleRequest(request.getBody().get(), context);
+	}
+
+	@FunctionName("getQuestionById")
+	public Question executeGetById(@HttpTrigger(name = "request", methods = {
+			HttpMethod.GET }, authLevel = AuthorizationLevel.ANONYMOUS, route = "questions/{questionId}") HttpRequestMessage<Optional<Question>> request,
+			ExecutionContext context, @BindingName("questionId") String questionId) {
+		context.getLogger().info("Retrieving question");
+		Question question = new Question();
+		question.setId(questionId);
+
+		return handleRequest(question, context);
+	}
+
+	@FunctionName("updateQuestion")
+	public Question executeUpdate(@HttpTrigger(name = "request", methods = {
+			HttpMethod.POST }, authLevel = AuthorizationLevel.ANONYMOUS, route = "questions/{questionId}") HttpRequestMessage<Optional<Question>> request,
```

Comment:
As per the spec, I think updates should be a PATCH
```suggestion
			HttpMethod.PATCH }, authLevel = AuthorizationLevel.ANONYMOUS, route = "questions/{questionId}") HttpRequestMessage<Optional<Question>> request,
```

Reply:
the azure functions HttpMethod java class does not seem to suppport PATCH, so I made it a PUT and modified the functions to update the non null fields when the function is called. We can talk about the best way to approach this

Reply:
:+1: I'm happy with this approach :)